src/routes/v0/stream/users.rs

```diff
@@ -82,18 +89,37 @@ pub async fn stream_users_handler(
                         .to_string(),
                 })
             }
+            UserStreamSource::Influencers => match query.source_reach {
+                None => (),
+                Some(_) => {
+                    return Err(Error::InvalidInput {
+                        message:
+                            "source_reach query param must be provided for source 'influencers' with a user_id"
+                                .to_string(),
+                    })
+                }
+            },
             _ => (),
         }
+    } else if source == UserStreamSource::Influencers && query.source_reach.is_none() {
+        return Err(Error::InvalidInput {
+                message:
+                    "source_reach query param must be provided for source 'influencers' when you pass a user_id"
+                        .to_string(),
+            });
     }
 
-    match UserStream::get_by_id(
-        query.user_id.as_deref(),
-        query.viewer_id.as_deref(),
-        Some(skip),
-        Some(limit),
-        source.clone(),
-        query.depth,
-    )
+    match UserStream::get_by_id(&UserStreamInput {
+        user_id: query.user_id.clone(),
+        viewer_id: query.viewer_id,
+        skip: Some(skip),
+        limit: Some(limit),
+        source: source.clone(),
+        source_reach: query.source_reach,
+        depth: query.depth,
+        timeframe: Some(timeframe),
+        preview: query.preview,
+    })
```

Comment:
This function exceeds the recommended maximum of 7 parameters, which may cause a `clippy` warning

Reply:
That's why it's a struct and the function is just getting one argument.

Reply:
ups! you are right, I miss that struct haahh